spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -429,6 +446,16 @@ def count
       it 'can use `super` to reference the parent context value' do
         expect(a_value).to eq("a string (modified)")
       end
+
+      it 'does not emit warnings on override' do
+        expect(RSpec).to_not receive(:warning)
+        RSpec.describe do
+          let(:foo) { 'first value' }
+          context 'in sub context' do
+            let(:foo) { 'second value' }
+          end
+        end
```

Comment:
As above, I think it's better to use the `output` matcher here:

``` ruby
it 'does not emit warnings on override' do
  expect {
    RSpec.describe do
      let(:foo) { 'first value' }
      context 'in sub context' do
        let(:foo) { 'second value' }
      end
    end
  }.to avoid_outputting.to_stdout.and avoid_outputting.to_stderr
end
```

* `avoid_outputting` is a negated form of `output`: https://github.com/rspec/rspec-core/blob/1e483f6de33a56bb93d8846644cf73e42d307596/spec/support/matchers.rb#L112
* If the implementation switched to using `Kernel#warn` (or some other mechanism), then the use of `expect(RSpec).to_not receive(:warning)` would be a false positive.  Really, we care that no output is emitted to stdout/stderr when loading this example group, regardless of the mechanisms for doing so.